src/docs/resources/compatibility.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Flutter compatibility policy
+description: How Flutter approaches the question of breaking changes
+---
+
+The Flutter team tries to balance the need for API stability with the
+need to keep evolving APIs to fix bugs, improve API ergonomics, and
+provide new features in a coherent manner.
+
+To this end, we have created a test registry where you can provide
+unit tests that we should run on every change to help us track down
+when changes would break existing applications. Our commitment is that
+we will not make any changes that break these tests without working
+with the developers of those tests to (a) determine if the change is
+sufficiently valuable and (b) provide fixes for the code so that the
```

Comment:
I'm a bit confused by this philosophy. It _sounds_ from a) as if the first person to contribute a specific test gets special treatment in determining the value of a breaking change: is that what you mean? I wonder if contributing a unit test automatically gives someone a good view of the impact of that break?

I think b) is the key here: we want to communicate that we'll never arbitrarily break a test without fixing the test (in the same way that we don't just break google3). So contributing a test both acts as (i) a canary for inadvertent breaks plus (ii) a path to a documented fix for a breaking change. Does that sound about right?

Reply:
not sure what you mean by "first person". Anyone can contribute their app's tests, and then we won't break those tests without talking to them first.

Reply:
Ah... it wasn't clear to me at first that you were asking folk to contribute a link to _their app test ***suite***_. Without reading through the actual contributor guide, it looked like you're asking for individual unit tests. I was reading it as implying that I might (for instance) contribute a test for `FloatingActionButton`, and based on that test, if we broke the FAB you'd ask me about whether the change was sufficiently valuable.

Reply:
Ah yeah that wasn't clear. I've clarified the text to say:

> To this end, we have created a test registry where you can provide unit tests for your own applications or libraries that we should run on every change to help us track down when changes would break existing applications.

Reply:
Great -- that totally resolves my confusion!